### DISCUSSION TITLE
Map Cache

### DISCUSSION THREAD

**Discussion by Thomas-Lerman** [pelicanmapping/osgearth#2659] [Support Q&A]
Questions concerning map cache while running in application using osgEarth. It seems like I knew some of these answers, but cannot seem to find them at this time, so apologize if already known:

1. To increase the speed of loading map data, especially Internet maps, can we programmatically increate the number of threads to use?
2. For the map caching, may I programmatically change the maximum cache size?
3. I am trying to understand how osgEarth names the cache folder for maps. At this point, I am seeing differences in naming betwee osgearth_cache and my application using osgEarth.

**Comment by gwaldron** [comment]
You can set the number of threads by calling
```c++
mapnode->getTerrainOptions().setConcurrency(n);
mapnode->getTerrainEngine()->dirtyTerrainOptions();
```

There is no maximum cache size in the default `filesystem` cache. If you use the optional `rocksdb` cache, you can use the `max_size_mb` option to set a target max size. This value is just a hint though, there is no guarantee it will be honored accurately.

The cache is meant to be a transient "black box" and the format subject to change from version to version.

**Comment by Thomas-Lerman** [comment]
Thank you very much. I was asking the last question as it seems the cache folder name is based upon at least the URI, image size, and probably more parameters. Since I am getting different cache folder names, I am presuming I do not have some parameter the same.

Is the setConcurrency() similar to the (old) loadingPolicy()?

**Reply by gwaldron** [reply]
Yeah, I'd say it is a simplified version of the old loading policy.

Again the cache is a black box, and there is no guarantee the format will be the same from version to version. You can't count on using a cache from an old version with a newer version.
